modules/dnn/src/cuda4dnn/csl/kernel_utils.hpp

```diff
@@ -0,0 +1,135 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_DNN_CUDA4DNN_CSL_KERNEL_UTILS_HPP
+#define OPENCV_DNN_CUDA4DNN_CSL_KERNEL_UTILS_HPP
+
+#include "error.hpp"
+#include "stream.hpp"
+#include "nvcc_defs.hpp"
+
+#include <utility>
+
+#ifdef __CUDACC__
+#include <cuda_runtime.h>
+#endif
+
+namespace cv { namespace dnn { namespace cuda4dnn { namespace csl {
+
+#ifdef __CUDACC__
+    struct execution_policy {
+        execution_policy(dim3 grid_size, dim3 block_size)
+            : grid{ grid_size }, block{ block_size }, sharedMem{ 0 }, stream{ 0 } { }
+
+        execution_policy(dim3 grid_size, dim3 block_size, std::size_t shared_mem)
+            : grid{ grid_size }, block{ block_size }, sharedMem{ shared_mem }, stream{ nullptr } { }
+
+        execution_policy(dim3 grid_size, dim3 block_size, const Stream& strm)
+            : grid{ grid_size }, block{ block_size }, sharedMem{ 0 }, stream{ StreamAccessor::get(strm) } { }
+
+        execution_policy(dim3 grid_size, dim3 block_size, std::size_t shared_mem, const Stream& strm)
+            : grid{ grid_size }, block{ block_size }, sharedMem{ shared_mem }, stream{ StreamAccessor::get(strm) } { }
+
+        dim3 grid;
+        dim3 block;
+        std::size_t sharedMem;
+        cudaStream_t stream;
+    };
+
+    template <class Kernel> inline
+    execution_policy make_optimal_policy(Kernel kernel, std::size_t sharedMem = 0, const Stream& stream = 0) {
```

Comment:
This doesn't make the optimal policy, just a reasonably good default. Consider naming this `make_default_policy` instead.

Reply:
Or even just `make_policy`.